pkg/providers/aws/provider_redis.go

```diff
@@ -271,11 +265,24 @@ func (p *RedisProvider) createElasticacheCluster(ctx context.Context, r *v1alpha
 	var replicationGroupClusters []elasticache.CacheCluster
 	for _, checkedCluster := range cacheClustersOutput.CacheClusters {
 		cluster := *checkedCluster
```

Comment:
NP : not related to this change, but if you are feeling it, I dont think we need to declare `cluster := *checkedCluster` we can just pass `checkedCluster.ReplicationGroupID` into the `SafeStringDereference`